x-pack/plugins/file_upload/server/capabilities.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CoreSetup } from 'kibana/server';
+import { checkFileUploadPrivileges } from './check_privileges';
+import { StartDeps } from './types';
+
+export const setupCapabilities = (
+  core: Pick<CoreSetup<StartDeps, unknown>, 'capabilities' | 'getStartServices'>
+) => {
+  core.capabilities.registerProvider(() => {
+    return {
+      fileUpload: {
+        show: true,
+      },
+    };
+  });
+
+  core.capabilities.registerSwitcher(async (request, capabilities, useDefaultCapabilities) => {
+    const isAnonymousRequest = !request.route.options.authRequired;
+    if (useDefaultCapabilities || isAnonymousRequest) {
+      return capabilities;
+    }
+    const [, { security }] = await core.getStartServices();
+
+    // Check the bare minimum set of privileges required to get some utility out of this feature
+    const { hasImportPermission } = await checkFileUploadPrivileges({
+      authorization: security?.authz,
+      request,
+      checkCreateIndexPattern: true,
+      checkHasManagePipeline: false,
+    });
```

Comment:
We have the flexibility to change these privilege checks to whatever we need...I just picked something to start the conversation. Should we also test that the user can create pipelines?